### PR TITLE
To Do List Best Practices Review

### PR DESCRIPTION
**Pull request by ryxtor**
Hello! 👋🏻 

This is a pull request to check if I'm following the [JavaScript Best Practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md), [DRY, KISS, and YAGNI](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/clean-code/lesson_DRY_KISS_YAGNI.md) and [HTML & CSS best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) rules.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,39 @@
-# To-Do-List-Best-Practices
-This is an exercise to follow the JS Best Practices
+# To Do List
+
```
Comment (richardoppiyo) [inline README.md:2]: I think your README.md file will be better having a single heading that describes the project. Kindly consider removing one heading and description tag.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  enIcon.src = enterIcon;
+  enIcon.setAttribute('id', 'enterInput');
+  enIcon.classList = 'enterInput';
+  enIcon.setAttribute('alt', 'enter-icon');
+  formContainer.appendChild(enIcon);
+};
+
+createEnterIcon();
+
+const runClass = new ToDoList();
```
Comment (richardoppiyo) [inline src/index.js]: To Follow the best JS practice, I suggest you consider having all the declarations at the top of your page for consistency purposes and to make your code even more readable

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+runClass.displaytdlist();
+const btnTask = document.getElementById('enterInput');
```
Comment (richardoppiyo) [inline src/index.js]: To Follow the best JS practice, I suggest you consider having all the declarations at the top of your page for consistency purposes and to make your code even more readable

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+btnTask.addEventListener('click', () => {
+  runClass.addTask();
+});
+const enterTask = document.getElementById('input');
```
Comment (richardoppiyo) [inline src/index.js]: To Follow the best JS practice, I suggest you consider having all the declarations at the top of your page for consistency purposes and to make your code even more readable

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+enterTask.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    runClass.addTask();
+  }
+});
+
+const delCompletedbtn = document.getElementById('clear-btn');
```
Comment (richardoppiyo) [inline src/index.js]: To Follow the best JS practice, I suggest you consider having all the declarations at the top of your page for consistency purposes and to make your code even more readable

File: src/modules/todoclass.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    this.description = description;
+    this.completed = completed;
+    this.index = index;
+  }
+
+  displaytdlist() {
+    this.tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+    if (ul.querySelectorAll('li')) {
+      Array.from(ul.querySelectorAll('li')).forEach((task) => {
```
Comment (richardoppiyo) [inline src/modules/todoclass.js]: You might consider using the shorthand arrow function syntax to make your code even more readable and clean
Consider:

Array.from(ul.querySelectorAll('li')).forEach((task) =>   ul.removeChild(task));

File: src/modules/todoclass.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+      li.className = 'task';
+      this.index += 1;
+      task.index = this.index;
+      li.setAttribute('id', `${this.index}`);
+
+      // Create Checkbox
+      const checkbox = document.createElement('input');
+      checkbox.setAttribute('type', 'checkbox');
+      checkbox.className = 'checkbox';
+      checkbox.addEventListener('change', () => {
```
Comment (richardoppiyo) [inline src/modules/todoclass.js]: You might consider using the shorthand arrow function syntax to make your code even more readable and clean
Consider:

checkbox.addEventListener('change', () =>   this.completeTask(checkbox.parentElement));

File: src/modules/todoclass.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+        inputText.classList.add('completeTask');
+      }
+
+      // Create Delete Button
+      const delIcon = new Image();
+      delIcon.src = deleteIcon;
+      delIcon.setAttribute('id', 'deleteTask');
+      delIcon.classList = 'deleteTask';
+      delIcon.setAttribute('alt', 'delete-icon');
+      delIcon.addEventListener('click', () => {
```
Comment (richardoppiyo) [inline src/modules/todoclass.js]: You might consider using the shorthand arrow function syntax to make your code even more readable and clean
Consider:

delIcon.addEventListener('click', () => this.removeTask(task.index));

**Review by richardoppiyo — CHANGES_REQUESTED** [review]
# Changes required ♻️ 

Kindly check the comment below and work on the changes. 😃 

Thanks

**Comment by ryxtor** [comment]
### Hello @richardoppiyo! :wave:

Thanks for your feedback, I've already made the changes. 😄

**Review by richardoppiyo — APPROVED** [review]

**Comment by richardoppiyo** [comment]
Hello 👋

Great work on implementing the changes 😄

**Pull request merged**